services/gtfs-rt-archiver-v3/gtfs_rt_archiver_v3/tasks.py

```diff
@@ -0,0 +1,114 @@
+import os
+from datetime import datetime
+
+import orjson
+import pendulum
+import structlog
+import typer
+from calitp.storage import AirtableGTFSDataRecord, download_feed
+from google.cloud import storage
+from huey import RedisExpireHuey
+from huey.registry import Message
+from huey.serializer import Serializer
+from requests import HTTPError
+
+from .metrics import (
+    FETCH_PROCESSING_TIME,
+    TASK_SIGNALS,
+    FETCH_PROCESSING_DELAY,
+    FETCH_PROCESSED_BYTES,
+)
+
+
+class PydanticSerializer(Serializer):
+    def _serialize(self, data: Message) -> bytes:
+        return orjson.dumps(data._asdict())
+
+    def _deserialize(self, data: bytes) -> Message:
+        # deal with datetimes manually
+        d = orjson.loads(data)
+        d["expires_resolved"] = datetime.fromisoformat(d["expires_resolved"])
+        d["kwargs"]["tick"] = datetime.fromisoformat(d["kwargs"]["tick"])
+        return Message(*d.values())
+
+
+huey = RedisExpireHuey(
+    name=f"gtfs-rt-archiver-v3-{os.environ['AIRFLOW_ENV']}",
+    expire_time=5,
+    results=False,
+    # serializer=PydanticSerializer(),
+    url=os.getenv("CALITP_HUEY_REDIS_URL"),
+    host=os.getenv("CALITP_HUEY_REDIS_HOST"),
+    port=os.getenv("CALITP_HUEY_REDIS_PORT"),
+    password=os.getenv("CALITP_HUEY_REDIS_PASSWORD"),
+)
+
+
+client = storage.Client()
+
+base_logger = structlog.get_logger()
+
+
+@huey.signal()
+def instrument_signals(signal, task, exc=None):
+    TASK_SIGNALS.labels(
+        record_name=task.kwargs["record"].name,
+        record_uri=task.kwargs["record"].uri,
+        record_feed_type=task.kwargs["record"].data,
+        signal=signal,
+        exc_type=type(exc) if exc else "",
+    ).inc()
+
+
+AUTH_KEYS = [
+    "AC_TRANSIT_API_KEY",
+    "AMTRAK_GTFS_URL",
+    "CULVER_CITY_API_KEY",
+    "MTC_511_API_KEY",
+    "SD_MTS_SA_API_KEY",
+    "SD_MTS_VP_TU_API_KEY",
+    "SWIFTLY_AUTHORIZATION_KEY_CALITP",
+]
```

Comment:
I do think it's a feature but I could be convinced otherwise. Mainly, I want a source code location that signals the API keys we will read in, and whether you need to add a missing key vs fixing an existing but misconfigured key. The k8s YAML could technically contain this but that would then require changing a kustomize file vs just source code in an image.

A more specific limitation is that I don't necessarily want to iterate over and/or load all secrets from Secret Manager; there may be a way to tag/categorize?